### PR TITLE
🐛 FIX: Redirect of filenames containing dots

### PR DESCRIPTION
**Pull request by chrisjsewell**
The use of `Path.with_suffix` (twice) incorrectly truncates file names,
e.g. `a.b.c.rst` -> `a.b.html`.
This fix only strips the known source suffixes, then appends `.html`.

### CODE REVIEW

File: sphinxext/rediraffe.py [hunk 1]
```diff
@@ -206,18 +214,22 @@ def build_redirects(app: Sphinx, exception: Union[Exception, None]) -> None:
...
+        redirect_to = src_redirect_to.parent / f"{redirect_to_name}.html"
 
         if type(app.builder) == DirectoryHTMLBuilder:
-            if redirect_from.name != "index":
-                redirect_from = redirect_from / "index"
-            if redirect_to.name != "index":
-                redirect_to = redirect_to / "index"
-
-        redirect_from = redirect_from.with_suffix(".html")
-        redirect_to = redirect_to.with_suffix(".html")
```
Comment (TheTripleV) [inline sphinxext/rediraffe.py:220]:
```py        
		redirect_from = redirect_from.with_suffix(redirect_from.suffix + ".html")
        redirect_to = redirect_to.with_suffix(redirect_to.suffix + ".html")
```

I would prefer this as a fix as it doesn't require an extra custom path function.

**Review by TheTripleV — CHANGES_REQUESTED** [review]
Thanks for finding this bug!

File: sphinxext/rediraffe.py (hunk 1 above)
Comment (chrisjsewell) [reply sphinxext/rediraffe.py:220]: Yeh but I'd say it's incorrect to just remove any suffix; it should only be ones that map to suffixes of sphinx source files, which in principle can also contain dots like `.myst.md`.
So, in the context of sphinx, using `.with_suffix` and `.suffix` is not a good idea

Comment (chrisjsewell) [reply sphinxext/rediraffe.py:220]: the code I added is exactly how sphinx does the suffix removal

Comment (chrisjsewell) [reply sphinxext/rediraffe.py:220]: for example here 😉 https://github.com/sphinx-doc/sphinx/blob/9f44ee4dd1761e12a25b63671dcc4b8ace1f2555/sphinx/directives/other.py#L120

Comment (chrisjsewell) [reply sphinxext/rediraffe.py:220]: do you still want me to change it?

Comment (TheTripleV) [reply sphinxext/rediraffe.py:220]: Ah, I hadn't considered multiple dots. This is fine then.

**Review by TheTripleV — APPROVED** [review]

**Comment by chrisjsewell** [comment]
thanks for the quick response @TheTripleV! If you could get it out in a release, that would be great 😬

**Pull request merged**